### PR TITLE
Fixes API permissions and resource schema validation

### PR DESCRIPTION
**Pull request by shangabl**
*Issue #, if available:*

*Description of changes:*

Minor fixes for permissions and schema validation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by HarshGandhi-AWS — APPROVED** [review]
LGTM

File: aws-iot-jobtemplate/resource-role.yaml [hunk 1]
```diff
@@ -23,10 +23,10 @@ Resources:
             Statement:
               - Effect: Allow
                 Action:
-                - "iot:CreateJobTemplates"
-                - "iot:DeleteJobTemplates"
+                - "iot:CreateJobTemplate"
+                - "iot:DeleteJobTemplate"
                 - "iot:DescribeJobTemplates"
-                - "iot:ListJobTemplates"
+                - "iot:ListJobTemplate"
```
Comment (BenChen-AWS) [inline aws-iot-jobtemplate/resource-role.yaml:29]: Action name should be `ListJobTemplates` and `DescribeJobTemplate`.

Comment (shangabl) [reply aws-iot-jobtemplate/resource-role.yaml]: Yikes, thanks for noticing that.

**Review by BenChen-AWS — APPROVED** [review]
Looks good to me.

**Pull request merged**